src/de/jost_net/JVerein/gui/parts/EditTreePart.java

```diff
@@ -0,0 +1,386 @@
+import org.eclipse.swt.events.TraverseEvent;
+import org.eclipse.swt.events.TraverseListener;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.graphics.Rectangle;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Event;
+import org.eclipse.swt.widgets.Listener;
+import org.eclipse.swt.widgets.Text;
+import org.eclipse.swt.widgets.Tree;
+import org.eclipse.swt.widgets.TreeItem;
+
+import de.jost_net.JVerein.gui.control.listener.TreeEditListener;
+import de.willuhn.datasource.BeanUtil;
+import de.willuhn.datasource.GenericIterator;
+import de.willuhn.jameica.gui.Action;
+import de.willuhn.jameica.gui.GUI;
+import de.willuhn.jameica.gui.parts.Column;
+import de.willuhn.jameica.gui.parts.TableChangeListener;
+import de.willuhn.jameica.gui.parts.TreePart;
+import de.willuhn.jameica.gui.parts.table.Feature;
+import de.willuhn.jameica.gui.parts.table.Feature.Context;
+import de.willuhn.jameica.gui.util.Color;
+import de.willuhn.jameica.messaging.StatusBarMessage;
+import de.willuhn.jameica.system.Application;
+import de.willuhn.logging.Logger;
+import de.willuhn.util.ApplicationException;
+
+/**
+ * Ein TreePart, bei dem Spalten editiert werden k�nnen. Dazu m�ssen die Spalten
+ * <code>editable</code> sein. Per
+ * <code>addChangeListener(TableChangeListener l)</code> kann ein Listener
+ * angeh�ngt werden, der bei �nderungen benachrichtigt wird.
+ */
+public class EditTreePart extends TreePart
+{
+  private TreeEditor editor;
+
+  private Tree tree;
+
+  private int selectedCol = -1;
+
+  private List<TableChangeListener> changeListeners = new ArrayList<>();
+
+  private List<TreeEditListener> editListeners = new ArrayList<>();
+
+  public EditTreePart(Object object, Action action)
+  {
+    super(object, action);
+  }
+
+  public EditTreePart(List<?> list, Action action)
+  {
+    super(list, action);
+  }
+
+  public EditTreePart(GenericIterator<?> list, Action action)
+  {
+    super(list, action);
+  }
+
+  @Override
+  protected Context createFeatureEventContext(Feature.Event e, Object data)
+  {
+    Context ctx = super.createFeatureEventContext(e, data);
+    tree = (Tree) ctx.control;
+    if (this.changeable)
+    {
+      this.editor = new TreeEditor(tree);
+      this.editor.horizontalAlignment = SWT.LEFT;
+      this.editor.grabHorizontal = true;
+
+      tree.addListener(SWT.MouseDown, new Listener()
+      {
+        public void handleEvent(Event e)
+        {
+          // Bearbeiten nur dann, wenn man mit der linken Maustaste klickt
+          if (e.button != 1)
+            return;
+
+          selectedCol = -1;
+
+          final TreeItem item = getSelectedItem(new Point(e.x, e.y),
+              tree.getColumnCount(), tree.getItems());
+
+          if (item == null || selectedCol == -1)
+            return;
+
+          final int index = selectedCol;
+
+          // Jetzt checken wir noch, ob die Spalte aenderbar ist
```

Comment:
Bitte bei der Schreibweise von Umlauten konsistent bleiben
```suggestion
          // Jetzt checken wir noch, ob die Spalte änderbar ist
```